### ISSUE TITLE
Point(NaN, NaN) no longer creates empty point with GEOS >= 3.13

### ISSUE THREAD

**Issue by jorisvandenbossche** [shapely/shapely#2143]
The behaviour of `Point(np.nan, np.nan)` changes depending on the GEOS version you have. For example, using in both cases shapely main but with a different version of GEOS:

```python
>>> shapely.geos_version
(3, 12, 1)
>>> shapely.Point(np.nan, np.nan)
<POINT EMPTY>
```
vs

```python
>>> shapely.geos_version
(3, 13, 0)
>>> shapely.Point(np.nan, np.nan)
<POINT (NaN NaN)>
```

This is known, for example we updated the tests and added a note about this in the `shapely.points` docstring in https://github.com/shapely/shapely/pull/1811. 
(although I am not entirely sure if it's actually only the WKT repr that changed on the GEOS side, or if also their internal model of how to create and track empty geometries changed)

While known (at least we tested for it), I do wonder if this is the best default option for users. 

Specifically for the `Point(..)` constructor, using NaNs was effectively a way to construct an empty point (although you can also do `Point()`), and that behaviour is breaking.

If that behaviour is breaking anyway, I am wondering if we shouldn't rather let it error by default when you pass NaNs, instead of returning a Point with NaN coordinates (and which is no longer considered as empty).

**Comment by sgillies** [comment]
@jorisvandenbossche whatever we do let's preserve, for example, `Point()` as the canonical way to create empty geometries.

I don't have a use case for `Point(np.nan, np.nan)`, but I feel like the new GEOS behavior is an improvement.

**Comment by jorisvandenbossche** [comment]
> whatever we do let's preserve, for example, `Point()` as the canonical way to create empty geometries

Yes, nothing is changing there (that's also something we control ourselves with a custom check in the constructor, without relying on GEOS)

**Comment by jorisvandenbossche** [comment]
cc @shapely/shapely-team other opinions or thoughts here? (_if_ we would like to change anything, I think it would be good to do that for 2.1, since we are upgrading the GEOS version for the wheels to 3.13, which includes this change in behaviour)
